### PR TITLE
All pages and components relating to displaying artwork on the Gallery 

### PR DESCRIPTION
**Pull request by Isaac-Best**
Gallery page, artwork page, mock data, artwork photos 
 
- Gallery page uses query parameters, default sort is most recent  
--- I removed the redundant code with the nested dynamic page however I couldn't get the pre-rendering with[ getStaticProps()](https://nextjs.org/docs/pages/building-your-application/data-fetching/get-static-props) to work at all so I had to simplify the logic quite a bit to get it to run the same with query parameters instead of routing parameters
--- sort by most votes is currently not implemented -> I do not know if its something we want /care for as the gallery will have to be re-written anyway **(just let me know if its a feature we want, I can add another client side query)**

- artwork page, lots of hard coding
--- if the artwork is too small I can mess around with tailwind some more to scale it up, or convert the images to svg's and size it up more naturally 

- the reason this is branched from gallery-01 
--- I had to delete some unused files for GitHub action to build properly, but I would like to keep them for later in phase 2 when we use some of nextJS pre-rendering features without digging in commit histories

### CODE REVIEW

File: src/app/gallery/page.tsx [hunk 1]
```diff
@@ -1,5 +1,95 @@
...
+      {/* yellow container */}
+      <div className="flex justify-between items-center md:flex-col py-10 px-5 bg-main-yellow">
+        <div className="text-left md:text-center">
+          <h1 className="font-bold text-4xl">Gallery</h1>
+          <p className="font-medium md:font-semi-bold text-lg">You can vote only once, so share with friends.</p>
+          <p className="font-medium md:font-semi-bold text-lg">Vote until <span className="font-extrabold"> June 29, 2014 </span> (12:00) midnight EST </p>
+        </div>
+        {/* image container */}
+        <div className="text-left md:py-5">
+          <Image src={gallery_image} alt="Previous artwork submission" />
```
Comment (Andrew-Dame) [inline src/app/gallery/page.tsx:38]: I'm not sure what this alt description means...could you make this a bit more clear?

File: src/app/gallery/page.tsx [hunk 2]
```diff
@@ -1,5 +1,95 @@
-export default function page() {
+"use client";
+import Image from "next/image";
+import Pagination from "../../../components/pagination/Pagination";
+import ArtworkCard from "../../../components/ArtworkCard";
+import gallery_image from "../../../public/gallery/gallery_image.svg";
+// eslint-disable-next-line
```
Comment (Andrew-Dame) [inline src/app/gallery/page.tsx:6]: Please make these comments more specific. `// eslint-disable-next-line react-hooks/rules-of-hooks` should work

File: src/app/artwork/[artworkId]/page.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  {"id": 1, "name": "Teruhito Sekine", "votes": 12, "url":"/gallery/artwork-1.png" },
+  {"id": 2, "name": "Teruhito Sekine", "votes": 15, "url":"/gallery/artwork-2.png"},
+  {"id": 3, "name": "Teruhito Sekine", "votes": 2,  "url":"/gallery/artwork-3.png" },
+  {"id": 4, "name": "Teruhito Sekine", "votes": 23, "url":"/gallery/artwork-4.png" },
+  {"id": 5, "name": "Teruhito Sekine", "votes": 12, "url":"/gallery/artwork-5.png" },
+  {"id": 6, "name": "Teruhito Sekine", "votes": 15, "url":"/gallery/artwork-6.png" },
+  {"id": 7, "name": "Teruhito Sekine", "votes": 2, "url":"/gallery/artwork-7.png" }
+];
+
+function getArtworkById(id:string) {
```
Comment (Andrew-Dame) [inline src/app/artwork/[artworkId]/page.tsx:21]: I don't see a need for the `artworks` variable here or `getArtworkById`.

If you import artworks, you can just grab the desired artwork with `find`:

```
import { artworks } from "../../../../mock/artworks";

...

const artwork = artworks.find(artwork => artwork.id === Number(id));
```

I see the potential optimization using modulus here; however, I don't think the gain is enough for the cost of space and increased complexity of adding an additional artworks array.

File: components/pagination/Pagination.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
+import Link from "next/link";
+import React from "react";
+import usePagination from "./usePagination";
+
+export type PaginationProps = {
+  totalItems: number
+  currentPage: number
+  renderPageLink: (page: number) => string
+  itemsPerPage?: number
```
Comment (Andrew-Dame) [inline components/pagination/Pagination.tsx:9]: Maybe don't make this an optional prop? There isn't much of an interface for this component, and `itemsPerPage` is a core configuration of this component in general.

**Review by Andrew-Dame — CHANGES_REQUESTED** [review]

**Comment by Andrew-Dame** [comment]
Could we improve the name of this PR please? Something just a bit more descriptive

File: components/ArtworkCard.tsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    url: string; 
+}
+
+const ArtworkCard = ({ id, name, votes, url }: ArtworkCardProps) => {
+  return (
+    <div id={id}>
+      <Link href={`/artwork/${id}`}>
+        <img className="w-4/8 rounded-lg" src={url} alt={name}/>
+      </Link>
+      {/* <p className="font-medium md:font-semi-bold text-lg">id: {id} </p> */}
```
[the remark this answers is not in the document]
Comment (Isaac-Best) [reply components/ArtworkCard.tsx:15]: For clarification you mean line 15 the commented out line only correct?

**Review by Andrew-Dame — APPROVED** [review]

**Pull request merged**